File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-swift/Sources/MailSdk/mailProviderPackageCatalog.swift
public struct MailProviderPackageCatalogEntry {
    public let providerKey: String
    public let pluginId: String
    public let driverId: String
    public let packageIdentity: String
    public let manifestPath: String
    public let readmePath: String
    public let sourcePath: String
    public let sourceSymbol: String
    public let builtin: Bool
    public let rootPublic: Bool
    public let status: String
    public let runtimeBridgeStatus: String
}

public enum MailProviderPackageCatalog {
    public static let entries: [MailProviderPackageCatalogEntry] = [
        .init(providerKey: "smtp", pluginId: "Mail-smtp", driverId: "sdkwork-mail-driver-smtp", packageIdentity: "MailSdkProviderSmtp", manifestPath: "Providers/MailSdkProviderSmtp/Package.swift", readmePath: "Providers/MailSdkProviderSmtp/README.md", sourcePath: "Providers/MailSdkProviderSmtp/Sources/MailSdkProviderSmtp/MailProviderSmtpPackageContract.swift", sourceSymbol: "MailProviderSmtpPackageContract", builtin: true, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved"),
        .init(providerKey: "imap", pluginId: "Mail-imap", driverId: "sdkwork-mail-driver-imap", packageIdentity: "MailSdkProviderImap", manifestPath: "Providers/MailSdkProviderImap/Package.swift", readmePath: "Providers/MailSdkProviderImap/README.md", sourcePath: "Providers/MailSdkProviderImap/Sources/MailSdkProviderImap/MailProviderImapPackageContract.swift", sourceSymbol: "MailProviderImapPackageContract", builtin: true, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved"),
    ]

public static func getMailProviderPackageByProviderKey(_ providerKey: String) -> MailProviderPackageCatalogEntry? {
        entries.first { $0.providerKey == providerKey }
    }

    public static func getMailProviderPackageByPackageIdentity(_ packageIdentity: String) -> MailProviderPackageCatalogEntry? {
        entries.first { $0.packageIdentity == packageIdentity }
    }

}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-kotlin/src/main/kotlin/com/sdkwork/Mail/metadata/mailProviderPackageCatalog.kt
package com.sdkwork.Mail.metadata

data class MailProviderPackageCatalogEntry(
    val providerKey: String,
    val pluginId: String,
    val driverId: String,
    val packageIdentity: String,
    val manifestPath: String,
    val readmePath: String,
    val sourcePath: String,
    val sourceSymbol: String,
    val builtin: Boolean,
    val rootPublic: Boolean,
    val status: String,
    val runtimeBridgeStatus: String,
)

object MailProviderPackageCatalog {
    val entries: List<MailProviderPackageCatalogEntry> = listOf(
        MailProviderPackageCatalogEntry("smtp", "Mail-smtp", "sdkwork-mail-driver-smtp", "com.sdkwork:Mail-sdk-provider-smtp", "providers/Mail-sdk-provider-smtp/build.gradle.kts", "providers/Mail-sdk-provider-smtp/README.md", "providers/Mail-sdk-provider-smtp/src/main/kotlin/com/sdkwork/Mail/provider/smtp/MailProviderSmtpPackageContract.kt", "MailProviderSmtpPackageContract", true, false, "future-runtime-bridge-only", "reserved"),
        MailProviderPackageCatalogEntry("imap", "Mail-imap", "sdkwork-mail-driver-imap", "com.sdkwork:Mail-sdk-provider-imap", "providers/Mail-sdk-provider-imap/build.gradle.kts", "providers/Mail-sdk-provider-imap/README.md", "providers/Mail-sdk-provider-imap/src/main/kotlin/com/sdkwork/Mail/provider/imap/MailProviderImapPackageContract.kt", "MailProviderImapPackageContract", true, false, "future-runtime-bridge-only", "reserved"),
    )

fun getMailProviderPackageByProviderKey(providerKey: String): MailProviderPackageCatalogEntry? =
        entries.firstOrNull { it.providerKey == providerKey }

    fun getMailProviderPackageByPackageIdentity(packageIdentity: String): MailProviderPackageCatalogEntry? =
        entries.firstOrNull { it.packageIdentity == packageIdentity }

}
